src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpUseAsSpanInsteadOfRangeIndexer.Fixer.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.NetCore.Analyzers.Performance;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Performance
+{
+    /// <summary>
+    /// CA1831, CA1832, CA1833: Use AsSpan or AsMemory instead of Range-based indexers when appropriate.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public sealed class CSharpUseAsSpanInsteadOfRangeIndexerFixer : UseAsSpanInsteadOfRangeIndexerFixer
+    {
+        protected override bool TrySplitExpression(
```

Comment:
I was going to suggest you to use language-agnostic `IOperation` based approach here, but it seems that `input[3..5]` actually generates an `OperationKind.None` node: https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IFromEndIndexOperation_IRangeOperation.cs#L52

@333fred Do we have a work item tracking the IOperation implementation for this case?

Reply:
I seem to remember a substantial amount was implemented, but we may have left it as `OperationKind.None` to prevent considering it "final." I don't remember what still needs to be implemented, if anything.

Reply:
I've renamed that issue to clarify that we need to confirm the implemention, not just add tests